osxkeychain/osxkeychain_darwin.go

```diff
@@ -69,7 +69,14 @@ func (h Osxkeychain) Delete(serverURL string) error {
 	errMsg := C.keychain_delete(s)
 	if errMsg != nil {
 		defer C.free(unsafe.Pointer(errMsg))
-		return errors.New(C.GoString(errMsg))
+		goMsg := C.GoString(errMsg)
+		if goMsg == errCredentialsNotFound {
+			return credentials.NewErrCredentialsNotFound()
+		}
+		if goMsg == errInteractionNotAllowed {
+			return ErrInteractionNotAllowed
+		}
+		return errors.New(goMsg)
```

Comment:
Maybe better with switch case?

```go
switch goMsg := C.GoString(errMsg); {
case errCredentialsNotFound:
	return credentials.NewErrCredentialsNotFound()
case errInteractionNotAllowed:
	return ErrInteractionNotAllowed
default:
	return errors.New(goMsg)
}
```

Reply:
Yeah, I took this from existing code, but was considering as well; let me update 👍